### PR TITLE
feat(performance): Add support for SSR streaming 

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Feat**

This PR solves [DIA-926]

### Description

In attempt to improve performance of initial baseline metrics, this adds SSR streaming to force, via React and styled-components. The implementation was fairly simple, and something we can toggle on / off with an ENV var:
- `ENABLE_SSR_STREAMING`

If enabled, we create a node stream during asset collection and pass that through, in the same way we currently pass the html through and down to express. And at the express level, instead of rendering and sending html, we render the html template, and then send it down as a stream. If ENV var is off, then fallback to normal send all the html at once behavior. 

One important thing to note! This react streaming API has been deprecated, and we'll need to swap it in with the new one once react 18 is in. That will pose some challenges, since styled-components + streaming + react 18 doesn't quite play well together. There are approaches though; just gotta make it work right. So before we celebrate, this might give us a temporary bump that might need to be reverted post upgrade. 

Example of streaming with really slow connection: 

https://github.com/user-attachments/assets/4d6bc6e4-ae16-4c7c-bc02-d53f2f551295

cc @artsy/diamond-devs 

[DIA-926]: https://artsyproduct.atlassian.net/browse/DIA-926?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/System/Router/renderServerApp.tsx [hunk 1]
```diff
@@ -78,5 +82,39 @@ export const renderServerApp = ({
 
   const statusCode = getENV("statusCode") ?? code
 
-  res.status(statusCode).render(`${PUBLIC_DIR}/html.ejs`, options)
+  res
```
Comment (araujobarret) [inline src/System/Router/renderServerApp.tsx:85]: non-blocker: do you think switch(true) would be better to read it here? It might be imo

**Review by araujobarret — APPROVED** [review]
Looks good, excited about this one!

File: src/System/Router/renderServerApp.tsx (hunk 1 above)
Comment (damassi) [reply src/System/Router/renderServerApp.tsx:87]: This works with express's funky callback style, i think. We'll have to revisit this when we get to the React 18 upgrade as this form of streaming is no longer supported. So this is all likely just temporary.

**Review by mzikherman — DISMISSED** [review]
I don't have much/any expertise here to really 'critique' the code, but pulling the branch down locally and browsing - nothing feels broken, locally i'm browsing a bunch of pages just fine.

Since it's behind a feature flag, we can merge it and enable on staging, let Integrity run, and so forth.

File: src/System/Router/renderServerApp.tsx (hunk 1 above)
Comment (mzikherman) [reply src/System/Router/renderServerApp.tsx:87]: What are some things to watch out for in production, especially metrics-wise, as a result of this once it lands and is enabled? Is it that, we should see maybe no change or only a very slight improvement on more modern devices w/ snappy connections, but a larger improvement on the lower powered ones?

Comment (damassi) [reply src/System/Router/renderServerApp.tsx:87]: Basically TTFB, total blocking time, ICP -- anything that measures how fast something hits the user visually.

Comment (damassi) [reply src/System/Router/renderServerApp.tsx:87]: The only way we can make this ultimately worthwhile would be to refactor our router in such a way as to ship the layout right away -- and then fetch and render subcontent. In effect, enable waterfall behavior. Not sure if its worth it to satisfy the bots when, at least on desktop, most of our pages on google search console are 🟢.

**Comment by MounirDhahri** [comment]
Impressive!

**Pull request merged**